### PR TITLE
Jerazo feature coping up opensoure features

### PR DESCRIPTION
**Pull request by jerazo**
- 32bit unique id system 
- validations and input standards
- recaptcha
- loggin attemps
- logs
- stop logging wrong passwords in LoginAttemps

### CODE REVIEW

File: app/API/V1/Entities/User.php [hunk 1]
```diff
@@ -40,46 +44,119 @@
...
+     */
+    public $identificationKey;
+
+    /**
+     * @ORM\Column(type="string", name="instadat_uuid", nullable=true)
      * @Gedmo\Versioned
+     * @var string $instadatUUid
      */
-    protected $name;
+    protected $instadatUUid;
```
Comment (tempestwf) [inline app/API/V1/Entities/User.php]: @jerazo Can we change this in open source and on the client branch to $thirdPartyUuid? That way they can stay in sync, with out being specific to either project.

File: tests/LoginAttemptTest.php [hunk 2]
```diff
@@ -79,31 +87,7 @@ public function testEmailVerification():void
 
             /** Try to log in with wrong password **/
             $response = $this->json('POST', '/auth/authenticate', ['email' => $userResult["email"], 'password' => 'wrong password']);
-            $response->assertResponseStatus(403);
```
Comment (tempestwf) [inline tests/LoginAttemptTest.php:82]: @jerazo should these tests have been removed?

File: app/API/V1/Repositories/LoginAttemptRepository.php [hunk 3]
```diff
@@ -152,36 +152,24 @@ private function getAttemptStatus(LoginAttempt $attempt)
 
         /** Check for attempts */
         if ($max_partial_lock && $attempt->getCount() >= $max_partial_lock) {
```
Comment (tempestwf) [inline app/API/V1/Repositories/LoginAttemptRepository.php:154]: @jerazo are all the account lock combinations tested? I see there was some big changes here, so I wanted to make sure they were all still supported, and working

**Review by tempestwf — CHANGES_REQUESTED** [review]

File: app/API/V1/Entities/User.php (hunk 1 above)
Comment (jerazo) [reply app/API/V1/Entities/User.php]: fixed

File: app/API/V1/Repositories/LoginAttemptRepository.php (hunk 3 above)
Comment (jerazo) [reply app/API/V1/Repositories/LoginAttemptRepository.php:154]: Yes, we have all the lock cases supported with such smaller code.

File: tests/LoginAttemptTest.php (hunk 2 above)
Comment (jerazo) [reply tests/LoginAttemptTest.php:82]: This is fine as if we do test the duration of elapse then the unit test will take much longer and might kill timeout.

**Review by tempestwf — APPROVED** [review]

**Pull request merged**